seqr/management/tests/load_rna_seq_tests.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+import mock
+
+from django.core.management import call_command
+from django.test import TestCase
+
+from seqr.models import Sample, RnaSeqOutlier
+from seqr.views.utils.orm_to_json_utils import get_json_for_rna_seq_outliers
+
+RNA_FILE_ID = 'tmp_-_2021-03-01T00:00:00_-_test_data_manager_-_new_muscle_samples.tsv.gz'
+EXISTING_SAMPLE_GUID = 'S000150_na19675_d2'
+
+class LoadRnaSeqTest(TestCase):
+    fixtures = ['users', '1kg_project', 'reference_data']
+
+    @mock.patch('seqr.views.apis.data_manager_api.ANALYST_PROJECT_CATEGORY', 'analyst-projects')
+    @mock.patch('seqr.management.commands.load_rna_seq.logger.info')
+    @mock.patch('seqr.management.commands.load_rna_seq.open')
+    @mock.patch('seqr.views.apis.data_manager_api.gzip.open')
+    def test_command(self, mock_gzip_open, mock_open, mock_logger):
+        mock_gzip_file = mock_gzip_open.return_value.__enter__.return_value
+        mock_gzip_file.__next__.return_value = 'sampleID\tgeneID\tdetail\tpValue\tpadjust\tzScore\n'
+        mock_gzip_file.__iter__.return_value = [
+            'NA19675_D2\tENSG00000240361\tdetail1\t0.01\t0.13\t-3.1\n',
+            'NA19675_D2\tENSG00000240361\tdetail2\t0.01\t0.13\t-3.1\n',
+            'NA19675_D2\tENSG00000233750\tdetail1\t0.064\t0.0000057\t7.8\n',
+            'NA19675_D3\tENSG00000233750\tdetail1\t0.064\t0.0000057\t7.8\n',
+        ]
+        mock_open.return_value.__enter__.return_value.__iter__.return_value = ['NA19675_D2\tNA19675_1']
+
+        with self.assertRaises(ValueError) as e:
+            call_command('load_rna_seq', RNA_FILE_ID)
+        self.assertEqual(str(e.exception), 'Unable to find matches for the following samples: NA19675_D2, NA19675_D3')
+
+        with self.assertRaises(ValueError) as e:
+            call_command('load_rna_seq', RNA_FILE_ID, '--mapping-file', 'map.tsv')
+        self.assertEqual(str(e.exception), 'Unable to find matches for the following samples: NA19675_D3')
```

Comment:
The exception is not handled by the command codes but those of the function `load_rna_seq`. Therefore, maybe the exception shouldn't be test here.

Reply:
I wanted some level of sanity testing that the called function is behaving as expected. I think given that these tests are already written I'm not going to remove them so we can have lower test coverage